x-pack/plugins/security_solution/public/detections/containers/detection_engine/rules/use_rule_execution_events.tsx

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useQuery } from 'react-query';
+import { useAppToasts } from '../../../../common/hooks/use_app_toasts';
+import { fetchRuleExecutionEvents } from './api';
+import * as i18n from './translations';
+
+// TODO: https://github.com/elastic/kibana/pull/121644 clean up
+export const useRuleExecutionEvents = (ruleId: string) => {
+  const { addError } = useAppToasts();
+
+  return useQuery(
+    'ruleExecutionEvents',
+    async ({ signal }) => {
+      const response = await fetchRuleExecutionEvents({ ruleId, signal });
+      return response.events;
+    },
+    {
+      onError: (e) => {
+        // TODO: Should it be responsible for showing toasts?
+        // TODO: Change the title
+        addError(e, { title: i18n.RULE_AND_TIMELINE_FETCH_FAILURE });
```

Comment:
Thanks for all the plumbing around RuleExecutionEvents! 🙂 

As for toasts, looks like a lotta folks use `useAppToasts()` within data fetching hooks like this, so probably -- makes it easier to consume as well. With regards to privileges, I was wondering if we have any scenarios where a user could view a rule but not have privileges for the execution-log? RBAC for execution-log is based on access to the rule, correct? So only additional privilege logic necessary here would be for any reusability out of this hook.